Sources/ScipioKit/ProjectGenerator/ProjectGenerator.swift

```diff
@@ -0,0 +1,505 @@
+
+        let fileReferences = try target.sources.paths.map { sourcePath in
+            let group = try self.group(
+                for: sourcePath.parentDirectory,
+                parentGroup: targetGroup,
+                sourceRoot: target.sources.root
+            )
+            return try group.addFile(
+                at: sourcePath.toPath(),
+                sourceTree: .sourceRoot,
+                sourceRoot: sourceRoot.toPath()
+            )
+        }
+
+        let buildFiles: [PBXBuildFile] = fileReferences.map { reference in
+            return addObject(PBXBuildFile(file: reference))
+        }
+
+        let compilePhase = addObject(
+            PBXSourcesBuildPhase(
+                files: buildFiles
+            )
+        )
+
+        if !target.underlyingTarget.resources.isEmpty {
+            logger.warning(
+            """
+            🚧 \(target.name) has resources. However, resource support is currently partial yet.
+            You can't use `Bundle.module` using Scipio
+            """)
+        }
+
+        let resourcePhase = addObject(
+            try makeResourcePhase(for: target, targetGroup: targetGroup)
+        )
+
+        // TODO generate resource accessor to support Bundle.module
+        // https://github.com/apple/swift-package-manager/blob/29a16bc2dc0ef72b7044c1dc6236236e3d0120e0/Sources/Build/BuildPlan.swift#L806-L852
+
+        return PBXNativeTarget(name: target.c99name,
+                               buildConfigurationList: buildConfigurationList,
+                               buildPhases: [compilePhase, resourcePhase],
+                               product: productRef,
+                               productType: productType)
+    }
+
+    private func makeResourcePhase(for target: ResolvedTarget, targetGroup: PBXGroup) throws -> PBXResourcesBuildPhase {
+        guard let sourceRoot else {
+            throw Error.invalidPackage(packageName: package.name)
+        }
+
+        let resourcesReferences: [PBXFileReference] = try target.underlyingTarget.resources.reduce([]) { lists, resource in
+            let resourcesGroup = try targetGroup.addGroup(named: "Resources").first!
+
+            switch resource.rule {
+            case .process(let localization):
+                guard localization == nil else {
+                    throw Error.localizationNotSupported(targetName: target.name)
+                }
+            case .copy:
+                break
+            }
+
+            let files = try walk(resource.path)
+            return try files.map { file in
+                return try resourcesGroup.addFile(at: file.toPath(), sourceRoot: sourceRoot.toPath())
+            } + lists
+        }
+
+        let buildFiles = resourcesReferences.map { PBXBuildFile(file: $0) }
+
+        return PBXResourcesBuildPhase(files: buildFiles)
+    }
+
+    private struct ClangTargetSettingResult {
+        var moduleMapPath: AbsolutePath?
+    }
```

Comment:
Is this struct really required?

Reply:
I feel the return value of `applyClangTargetSpecificSettings` seems to be ambiguous.
So I wrap `AbsolutePath` in a named struct.